ste/folderCreationTracker.go

```diff
@@ -9,6 +9,78 @@ import (
 	"github.com/Azure/azure-storage-azcopy/v10/common"
 )
 
+type TrieNode struct {
+	children map[rune]*TrieNode
+	value    *uint32
+	isEnd    bool
+}
+
+type Trie struct {
+	root *TrieNode
+}
+
+func NewTrie() *Trie {
+	return &Trie{
+		root: &TrieNode{children: make(map[rune]*TrieNode)},
+	}
+}
+
+func (t *Trie) Insert(key string, value uint32) {
+	node := t.root
+	for _, char := range key {
+		if _, exists := node.children[char]; !exists {
+			node.children[char] = &TrieNode{children: make(map[rune]*TrieNode)}
+		}
+		node = node.children[char]
```

Comment:
We are searchig the map twice here once in if condition and then to get the node back. I think we can do this in one shot only during if check